src/Workleap.Extensions.OpenAPI.Analyzers/EnforceTypedResultsReturnAnalyzer.cs

```diff
@@ -0,0 +1,94 @@
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Workleap.Extensions.OpenAPI.Analyzers.Internals;
+
+namespace Workleap.Extensions.OpenAPI.Analyzer;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public class EnforceTypedResultsReturnAnalyzer : DiagnosticAnalyzer
+{
+    private static readonly DiagnosticDescriptor EndpointReturnTypedResult = new(
+        id: RuleIdentifiers.HasTypedResultsUsage,
+        title: "Validate endpoint return type",
```

Comment:
Maybe more explicit title: Enforce strongly typed endpoint response. (same with message Format we could recommend what to use instead